modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenProperty.java

```diff
@@ -61,6 +61,7 @@
     public boolean isReadOnly;
     public boolean isWriteOnly;
     public boolean isNullable;
+    public boolean isOptionalNullable;
```

Comment:
@bkabrda Seems like this is only set or processed in the Java client generator and you're using the vendor extension in the mustache template instead, right?

My take is to remove it and use only the vendor extension

Reply:
The reason why I put it here is that I have a strong belief that most of the other templates have the same problem that I'm trying to address in this PR for Java. So that made me thing that these other templates will benefit from having access to this variable in the rendering context, as the solution will usually look somewhat similar to what I did here. I do believe that for a large number of languages, we will need to specialcase these not-required-and-nullable fields, so I felt it was appropriate to make this general.

If that explanation is not sufficient, I can definitely make it Java-specific and we can make this general as soon as we find that another language needs it.

Reply:
Is `isOptionalNullable` used in the Java client template as part of this PR? I couldn't find it.

What about using `{{^required}}{{#isNullable}} ... {{/isNullable}}{{/required}}` for the time being?

Reply:
So yes and no. Currently, the `isJacksonOptionalNullable` variable is used in the Java client templates. The reason for that is that I couldn't find an easy way to make this work with gson - so my initial PR used `isOptionalNullable`, but then I moved to using `vendorExtensions.isJacksonOptionalNullable` (which is basically `isOptionalNullable && jackson`).

It is possible to use `{{^required}}{{#isNullable}} ... {{/isNullable}}{{/required}}`, but the problem is that sometimes you need an "else clause" to that (IOW a reverse of that condition), which is impossible to do in mustache AFAICS - you'd need to do explicitly do something else for all the other three combinations of possible values of `required` and `isNullable`.

So if this seems to be a problem for you, I can definitely drop this for now and only use `isJacksonOptionalNullable` and we can reintroduce this whenever it's used in another set of templates to share the functionality.

Reply:
I understand your intention to make it easier for all generators to easily support optional nullable field but given that it's not used at the moment, I suggest we remove it from this PR for the time being to keep this PR simple (easier to get it merged)

(btw, when adding a new property to the Codegen Property class, you will also need to update other methods, e.g. https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenProperty.java#L538)

Reply:
Ok, I'll remove it. 👍